### PR TITLE
[DEV-778] added basic wallet-setup generating and restoring wallet fr…

### PR DESCRIPTION
**Pull request by Krysh90**
…om a seed

### CODE REVIEW

File: core/shared/util.ts [hunk 1]
```diff
@@ -95,4 +96,15 @@ export class Util {
       muted = isPassword;
     });
   }
+
+  // --- JELLYFISH UTIL --- //
+  static readNetwork(): Network {
```
Comment (davidleomay) [inline core/shared/util.ts:102]: Use this for signature verification (`crypto.ts`)? 😉

File: core/cold-wallet/wallet/cold-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  private wallet: JellyfishWallet<WhaleWalletAccount, WalletHdNode> | undefined;
+
+  constructor(seed: string[], network: Network) {
+    this.seed = seed;
+    this.network = network;
+    this.client = new WhaleApiClient({
+      url: process.env.OCEAN_URL,
+      version: process.env.OCEAN_VERSION,
+      network: network.name,
+    });
```
Comment (davidleomay) [inline core/cold-wallet/wallet/cold-wallet.ts]: Why do we have a whale client on a cold wallet?

File: core/shared/util.ts (hunk 1 above)
Comment (Krysh90) [reply core/shared/util.ts:102]: would be a good idea :D will adapt

File: core/cold-wallet/wallet/cold-wallet.ts (hunk 2 above)
Comment (Krysh90) [reply core/cold-wallet/wallet/cold-wallet.ts]: good point should have added, that this need to move out of cold-wallet, as soon as MN manager is able to do that :)

Will add a comment for this

**Review by davidleomay — APPROVED** [review]

**Pull request merged**